sunpy/map/mapbase.py

```diff
@@ -1673,7 +1696,7 @@ def peek(self, draw_limb=False, draw_grid=False,
 
     @u.quantity_input
     def plot(self, annotate=True, axes=None, title=True,
-             clip_interval: u.percent = None, **imshow_kwargs):
+             clip_interval: u.percent = None, plot_settings=None, **imshow_kwargs):
```

Comment:
My interpretation of @dstansby's comment in #3415 was that we should support `interpolation` and `origin` as explicit kwargs to `plot` (and consequentially `peek`). This just means setting the `imshow_args` in `plot` a bit more explicitly.

Reply:
I am thinking of defining `imshow_args` as below:
```
imshow_args = copy.deepcopy(imshow_kwargs)
if self.plot_settings is not None:
	imshow_args.update(self.plot_settings)
```

Also in some scenarios like https://github.com/sunpy/sunpy/blob/6a5fab2591c0e12f29c0c622f5f52f06f4eb3efd/sunpy/map/compositemap.py#L425 the `cmap` property is still accessed as `"cmap": m.plot_settings['cmap']`, and it expects to provide for a default value to it.
Do all the instances like these need to be edited?

Reply:
@wtbarnes?

Reply:
You don't want to initialize `imshow_args` with `imshow_kwargs` as we want to be able to override any defaults with the keyword arguments passed to `plot`. Additionally, we want to move away from any internal use of `plot_settings`.

The fix is even simpler I think. Instead of initializing `imshow_args` with `plot_settings`, just create a dictionary explicitly from `norm`, `cmap`, etc.

```python
imshow_args = {
    'cmap': self.cmap,
    'norm': self.norm,
    'interpolation': interpolation,
    'origin': origin,
}
```

where `interpolation` and `origin` are now explicit kwargs for `plot` with default values of 'nearest' and 'lower', respectively.

Reply:
Actually, for the sake of simplicity, just initialize `imshow_args` as,

```python
imshow_args = {
    'cmap': self.cmap,
    'norm': self.norm,
    'interpolation': 'nearest',
    'origin': 'lower',
}
```

and ignore my comment about passing additional keyword arguments to plot. These can just be absorbed into `imshow_kwargs`.

Reply:
Okay! So I'm defining `imshow_args` as below:
```python
if 'interpolation' not in imshow_kwargs:
	interpolation = 'nearest'
else:
	interpolation = imshow_kwargs['interpolation']
        
if 'origin' not in imshow_kwargs:
	origin = 'lower'
else:
	origin = imshow_kwargs['origin']
        
imshow_args = {
	'cmap': self.cmap,
	'norm': self.norm,
	'interpolation': interpolation,
	'origin': origin,
}
        
```

Reply:
Also, what about this?https://github.com/sunpy/sunpy/blob/6a5fab2591c0e12f29c0c622f5f52f06f4eb3efd/sunpy/map/compositemap.py#L425
Do I need to remove all these instances by `cmap = self.cmap`? Or `plot_settings` need to be handled for this?

Reply:
All the checking of `imshow_kwargs` is  unnecessary. Anything passed in with `imshow_kwargs` will override the values in `imshow_args` in [this line](https://github.com/sunpy/sunpy/blob/6a5fab2591c0e12f29c0c622f5f52f06f4eb3efd/sunpy/map/mapbase.py#L1753).

Reply:
Yeah I saw that later :sweat_smile:

Reply:
Yes, all references to `plot_settings['cmap']` should be replaced with `self.cmap` (same with `norm`). I think you got most of them?

Reply:
Yeah! But then also build fails :disappointed:
I'm unsure of what is causing the issue now

Reply:
The problem seems to be with the gallery examples. If you can, try to build the docs (and gallery) locally (https://docs.sunpy.org/en/latest/dev_guide/documentation.html) to see which examples are causing the problems. The CI seems to be stopping every time it hits a gallery example that fails.